Fabric.Authorization.API/Modules/UsersModule.cs

```diff
@@ -270,6 +303,58 @@ public class UsersModule : FabricModule<User>
                     HttpStatusCode.NotFound);
             }
         }
+
+        private async Task<dynamic> GetUserRoles()
+        {
+            this.RequiresClaims(AuthorizationReadClaim);
+            var roleUserRequest = this.Bind<RoleUserRequest>();
+            try
+            {
+                var roles =
+                    await _userService.GetRolesForUser(roleUserRequest.SubjectId, roleUserRequest.IdentityProvider);
+                return roles.Select(r => r.ToRoleApiModel());
+            }
+            catch (NotFoundException<User>)
+            {
+                return CreateFailureResponse(
+                    $"User with SubjectId: {roleUserRequest.SubjectId} and Identity Provider: {roleUserRequest.IdentityProvider} was not found",
+                    HttpStatusCode.NotFound);
+            }
+        }
+
+        private async Task<dynamic> AddRolesToUser(dynamic param)
+        {
+            var apiRoles = this.Bind<List<RoleApiModel>>();
+            var errorResponse = await ValidateRolesToAdd(apiRoles);
+            if (errorResponse != null)
+            {
+                return errorResponse;
+            }
+

```

Comment:
Verify `subjectId` and `identityProvider` are not null?

Reply:
Same comment as above, since the parameters are in the route, we wouldn't route to this function if they are null.